gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/catalog/messages/GfxdSystemProcedureMessage.java

```diff
@@ -1062,7 +1066,65 @@ String getSQLStatement(Object[] params) throws StandardException {
             .append(params[2]).append(')').toString();
       }
     },
-    
+
+    setLogLevel {
+
+      @Override
+      boolean allowExecution(Object[] params) {
+        // so allowing log level to be set on all nodes including locator
+        return true;
+      }
+
+      @Override
+      public void processMessage(Object[] params, DistributedMember sender)
+          throws StandardException {
+        try {
+          String logClass = (String) params[0];
+          Level level = org.apache.log4j.Level.toLevel((String) params[1]);
+          SanityManager.DEBUG_PRINT(GfxdConstants.TRACE_SYS_PROCEDURES,
+                  "GfxdSystemProcedureMessage: setting log level for class '" + logClass
+                          + "' to " + level);
+          if (logClass.equals("")) {
+            // sets the log level for the root logger and the GFXD bridge
+            LogManager.getRootLogger().setLevel(level);
+            if (InternalDistributedSystem.getLoggerI18n() instanceof GFToSlf4jBridge) {
+              ((GFToSlf4jBridge) InternalDistributedSystem.getLoggerI18n()).setLevelForLog4jLevel(level);
```

Comment:
Thinking about it, will be better to use Gem level names since that is what log-level uses in lead/server configuration else it can cause confusion. Use LogWriterImpl.levelNameToCode and then don't need the conversion in setLevelForLog4jLevel

Reply:
I didn't get this completely. how do we set the log4j level with a gemxd level?  Do we create a mapping other way round. BTW, our log4j.properties uses log4j levels and not store levels.

Reply:
Right, log4j properties has log4j names but log-level has Gem so there is an inconsistency already. Also procedures cannot yet be executed from Spark layer. @kneeraj can you track the issues to add system procedure execution from Spark and log-level inconsistency?